feature-libs/product-multi-dimensional/selector/components/guards/product-multi-dimensional-selector.guard.ts

```diff
@@ -75,24 +75,30 @@ export class ProductMultiDimensionalSelectorGuard {
   protected findValidProductCodeAndReturnUrlTree(
     product: Product
   ): Observable<boolean | UrlTree> {
-    const variantOptions = product.variantOptions ?? [];
-    const results: VariantOption | undefined = variantOptions.find(
+    const validVariantCode: string | undefined = product.variantOptions?.find(
       (variant: VariantOption) => variant.stock && variant.stock.stockLevel
-    );
-    const productCode = results ? results.code : variantOptions[0].code;
-    return productCode
-      ? this.productService.get(productCode, ProductScope.LIST).pipe(
-          filter(isNotUndefined),
-          take(1),
-          map((multiDimensionalProduct: Product) => {
-            return this.router.createUrlTree(
-              this.semanticPathService.transform({
-                cxRoute: 'product',
-                params: multiDimensionalProduct,
-              })
-            );
-          })
-        )
-      : of(false);
+    )?.code;
+
+    const fallbackProductCode = product.variantOptions?.length
+      ? product.variantOptions[0]?.code
+      : '';
+
+    if (validVariantCode || fallbackProductCode) {
+      const productCode = validVariantCode ?? fallbackProductCode;
+
+      return this.productService.get(productCode, ProductScope.LIST).pipe(
+        filter(isNotUndefined),
+        take(1),
+        map((multiDimensionalProduct: Product) => {
+          return this.router.createUrlTree(
+            this.semanticPathService.transform({
+              cxRoute: 'product',
+              params: multiDimensionalProduct,
+            })
+          );
+        })
+      );
+    }
+    return of(false);
   }
```

Comment:
`minor:` consider simplifying this method by extracting some logic to a private method and inverting the `if` statement condition:
```ts
  protected findValidProductCodeAndReturnUrlTree(
    product: Product
  ): Observable<boolean | UrlTree> {
    const validVariantCode = this.getValidVariantCode(product);
    const fallbackProductCode = this.getFallbackProductCode(product);

    const productCode = validVariantCode ?? fallbackProductCode;

    if (productCode) {
      return this.productService.get(productCode, ProductScope.LIST).pipe(
        filter(isNotUndefined),
        take(1),
        map((multiDimensionalProduct: Product) =>
          this.router.createUrlTree(
            this.semanticPathService.transform({
              cxRoute: 'product',
              params: multiDimensionalProduct,
            })
          )
        )
      );
    }
    return of(false);
  }

  private getValidVariantCode(product: Product): string | undefined {
    return product.variantOptions?.find(
      (variant: VariantOption) => variant.stock && variant.stock.stockLevel
    )?.code;
  }

  private getFallbackProductCode(product: Product): string | undefined {
    return product.variantOptions?.length
      ? product.variantOptions[0]?.code
      : '';
  }
```

Reply:
thanks, looks better. I've implemented it.